modules/apps/commerce/commerce-order-content-web/src/main/resources/META-INF/resources/js/multishipping/Multishipping.tsx

```diff
@@ -337,17 +356,41 @@ const Multishipping = ({
 		}));
 	}, []);
 
-	const handleRowUpdate = useCallback((orderItem: IOrderItem) => {
-		setFormattedOrderItems((prevState) => {
-			return prevState.map((item) => {
-				if (item.id === orderItem.id) {
-					return orderItem;
+	const handleRowUpdate = useCallback(
+		async (orderItem: IOrderItem, saveFullOrder: boolean = false) => {
+			const originalFormattedOrderItems = {...formattedOrderItems};
```

Comment:
[NB] Consider `[structuredClone](https://developer.mozilla.org/en-US/docs/Web/API/Window/structuredClone)` to avoid dirty-referencing, especially if the object is nested and contains other complex data structures.